### PR TITLE
Remove PrivateSentrySDKOnly.h from umbrella header

### PR DESCRIPTION
**Pull request by brustolin**
Remove PrivateSentrySDKOnly.h from umbrella headers to prevent users from use it by mistake.

_#skip-changelog_

### CODE REVIEW

**Comment by philipphofmann** [comment]
We need to inform all hybrid SDKs because they have to make some changes after we merge this PR. Maybe also be worth asking them in advance if they are okay with the change.

**Comment by brustolin** [comment]
>We need to inform all hybrid SDKs because they have to make some changes after we merge this PR. Maybe also be worth asking them in advance if they are okay with the change.

Yes, I open this https://github.com/getsentry/team-mobile/issues/66 to address the problem.
But I may fix this in the hybrids SDKs myself.

File: Sentry.podspec [hunk 1]
```diff
@@ -30,7 +30,7 @@ Pod::Spec.new do |s|
 
   s.subspec 'Core' do |sp|
       sp.source_files = "Sources/Sentry/**/*.{h,hpp,m,mm,c,cpp}",
-        "Sources/SentryCrash/**/*.{h,hpp,m,mm,c,cpp}", "Sources/SentrySwift/**/*.{swift}"
+        "Sources/SentryCrash/**/*.{h,hpp,m,mm,c,cpp}", "Sources/Swift/**/*.{swift}"
```
Comment (philipphofmann) [inline Sentry.podspec:33]: `h`: Why do we have such a change in this PR?

Comment (brustolin) [reply Sentry.podspec:33]: Good question.

I think is a merge problem.

**Comment by brustolin** [comment]
Closing this in favor of #2349

**Pull request closed without merging**